### PR TITLE
Disconnect ble device if connect attempt fails to recover slot

### PR DESCRIPTION
**Pull request by bdraco**
Ensures the slot is recovered if the connection fails

### CODE REVIEW

**Comment by bdraco** [comment]
```
2022-10-25 10:49:38.483 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Connecting (attempt: 1, last rssi: -63)
2022-10-25 10:49:38.484 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] Checking if tracker_proxy_b can connect, available=True, ble_connections_free=1
2022-10-25 10:49:38.518 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] tracker_proxy_b: BLE connection limits: 0/1
2022-10-25 10:49:41.412 DEBUG (MainThread) [aiohomekit.controller.ble.controller] Timed out after 30 waiting for discovery with address B0EDEB4E-1103-5CDD-552D-92A54D2D8B30
2022-10-25 10:49:52.738 WARNING (MainThread) [aioesphomeapi.client] finished connect finally
2022-10-25 10:49:52.739 WARNING (MainThread) [aioesphomeapi.client] connect failed, disconnecting
2022-10-25 10:49:52.741 WARNING (MainThread) [aioesphomeapi.client] connect failed, finished disconnecting
2022-10-25 10:49:52.743 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Timed out trying to connect (attempt: 1, last rssi: -63)
2022-10-25 10:49:52.997 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Connecting (attempt: 2, last rssi: -63)
2022-10-25 10:49:52.998 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-25 10:49:55.000 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Timed out trying to connect (attempt: 2, last rssi: -63)
2022-10-25 10:49:55.253 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Connecting (attempt: 3, last rssi: -63)
2022-10-25 10:49:55.254 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-25 10:49:57.257 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Timed out trying to connect (attempt: 3, last rssi: -63)
2022-10-25 10:49:57.510 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Connecting (attempt: 4, last rssi: -63)
2022-10-25 10:49:57.511 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-25 10:49:59.516 DEBUG (MainThread) [bleak_retry_connector]  (C1:C7:27:7D:55:AB) - C1:C7:27:7D:55:AB: Timed out trying to connect (attempt: 4, last rssi: -63)
```


Even with the disconnect, I never see the slot becoming available again

**Comment by bdraco** [comment]
There is still a race
```
2022-10-30 12:03:14.206 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Connecting (attempt: 1, last rssi: -60)
2022-10-30 12:03:14.207 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth] Checking if tracker_proxy_b can connect, available=True, ble_connections_free=1
2022-10-30 12:03:14.231 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] tracker_proxy_b: BLE connection limits: 0/1
2022-10-30 12:03:20.943 ERROR (MainThread) [async_upnp_client.ssdp] Received error: [Errno 65] No route to host, transport: <_SelectorDatagramTransport fd=34 read=polling write=<idle, bufsize=0>>
2022-10-30 12:03:28.342 DEBUG (MainThread) [homeassistant.components.esphome.entry_data] Dispatching update with key (<class 'aioesphomeapi.model.SensorState'>, 1445088998): SensorState(key=1445088998, state=390.8479919433594, missing_state=False)
2022-10-30 12:03:30.461 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Timed out trying to connect (attempt: 1, last rssi: -60)
2022-10-30 12:03:30.715 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Connecting (attempt: 2, last rssi: -60)
2022-10-30 12:03:30.718 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-30 12:03:32.722 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Timed out trying to connect (attempt: 2, last rssi: -60)
2022-10-30 12:03:32.977 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Connecting (attempt: 3, last rssi: -60)
2022-10-30 12:03:32.978 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-30 12:03:34.980 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Timed out trying to connect (attempt: 3, last rssi: -60)
2022-10-30 12:03:35.233 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Connecting (attempt: 4, last rssi: -60)
2022-10-30 12:03:35.235 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-30 12:03:37.239 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Timed out trying to connect (attempt: 4, last rssi: -60)
2022-10-30 12:03:37.494 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Connecting (attempt: 5, last rssi: -60)
2022-10-30 12:03:37.495 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] tracker_proxy_b: Out of connection slots, waiting for a free one
2022-10-30 12:03:39.496 DEBUG (MainThread) [bleak_retry_connector] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46) - CB:86:15:F3:0A:14: Timed out trying to connect (attempt: 5, last rssi: -60)
```

**Comment by bdraco** [comment]
making progress but now have
```
2022-10-30 12:04:16.563 DEBUG (MainThread) [bleak_retry_connector] Bleak error calling <function restore_connection_and_resume.<locals>._async_restore_and_resume at 0x151d4ce50>, backing off: 0.1, retrying...
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 167, in _async_restore_and_resume
    return await func(self, *args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 521, in _process_disconnected_events_with_retry
    protocol_param = await self._get_all_protocol_params()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 848, in _get_all_protocol_params
    resp = await self._async_request(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 325, in _async_request
    return await self._async_request_under_lock(opcode, char, data, iid)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 344, in _async_request_under_lock
    pdu_status, result_data = await ble_request(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/client.py", line 72, in ble_request
    await _write_pdu(client, encryption_key, opcode, handle, iid, data, tid)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/client.py", line 99, in _write_pdu
    await client.write_gatt_char(handle, write, True)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/bleak/__init__.py", line 593, in write_gatt_char
    await self._backend.write_gatt_char(char_specifier, data, response)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 66, in _async_wrap_bluetooth_connected_operation
    return next(iter(done)).result()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 78, in _async_wrap_bluetooth_operation
    return await func(self, *args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 377, in write_gatt_char
    characteristic = self._resolve_characteristic(char_specifier)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 322, in _resolve_characteristic
    raise BleakError(f"Characteristic {char_specifier} was not found!")
bleak.exc.BleakError: Characteristic None was not found!

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/bleak_retry_connector/__init__.py", line 619, in _async_wrap_bluetooth_connection_error_retry
    return await func(*args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 176, in _async_restore_and_resume
    await self._async_restore_subscriptions()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 1031, in _async_restore_subscriptions
    await self._async_set_broadcast_encryption_key()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 633, in _async_set_broadcast_encryption_key
    await self._async_request_under_lock(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/pairing.py", line 344, in _async_request_under_lock
    pdu_status, result_data = await ble_request(
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/client.py", line 72, in ble_request
    await _write_pdu(client, encryption_key, opcode, handle, iid, data, tid)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/aiohomekit/controller/ble/client.py", line 99, in _write_pdu
    await client.write_gatt_char(handle, write, True)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/bleak/__init__.py", line 593, in write_gatt_char
    await self._backend.write_gatt_char(char_specifier, data, response)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 66, in _async_wrap_bluetooth_connected_operation
    return next(iter(done)).result()
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 78, in _async_wrap_bluetooth_operation
    return await func(self, *args, **kwargs)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 377, in write_gatt_char
    characteristic = self._resolve_characteristic(char_specifier)
  File "/Users/bdraco/home-assistant/venv/lib/python3.10/site-packages/homeassistant/components/esphome/bluetooth/client.py", line 322, in _resolve_characteristic
    raise BleakError(f"Characteristic {char_specifier} was not found!")
bleak.exc.BleakError: Characteristic None was not found!
```

**Comment by bdraco** [comment]
```
        if char.handle:
            endpoint = self.client.get_characteristic_by_handle(char.handle)
        else:
            endpoint = self.client.get_characteristic(char.service.type, char.type)
```

The endpoint is `None`

Should not be possible

File: aioesphomeapi/client.py [hunk 1]
```diff
@@ -485,6 +485,12 @@ def unsub() -> None:
         except asyncio.TimeoutError as err:
             unsub()
             raise TimeoutAPIError("Timeout waiting for connect response") from err
+        finally:
+            _LOGGER.warning("finished connect finally")
+            if not event.is_set():
+                _LOGGER.warning("connect failed, disconnecting")
+                await self.bluetooth_device_disconnect(address)
+                _LOGGER.warning("connect failed, finished disconnecting")
```
Comment (bdraco) [inline aioesphomeapi/client.py:493]: TODO: remove the testing loggers before marking ready

**Comment by bdraco** [comment]
2022-10-30 12:52:43.783 WARNING (MainThread) [aiohomekit.controller.ble.pairing] VOCOlinc-VS1-185620 [CB:86:15:F3:0A:14] (id=C0:3E:A4:C9:D1:46): Requesting data from char: <aiohomekit.model.characteristics.characteristic.Characteristic object at 0x28dc4ecb0>, handle: 44, endpoint: None


So the endpoint resolution is failing even though we have the handle

**Comment by bdraco** [comment]
```
2022-10-30 12:52:43.234 DEBUG (MainThread) [homeassistant.components.esphome.bluetooth.client] VOCOlinc-VS1-185620 - CB:86:15:F3:0A:14: Got services: ESPHomeBluetoothGATTServices(address=223776754305556, services=[BluetoothGATTService(uuid='00001800-0000-1000-8000-00805f9b34fb', handle=1, characteristics=[BluetoothGATTCharacteristic(uuid='00002a00-0000-1000-8000-00805f9b34fb', handle=3, properties=10, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002a01-0000-1000-8000-00805f9b34fb', handle=5, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002a04-0000-1000-8000-00805f9b34fb', handle=7, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00002aa6-0000-1000-8000-00805f9b34fb', handle=9, properties=2, descriptors=[])]), BluetoothGATTService(uuid='00001801-0000-1000-8000-00805f9b34fb', handle=10, characteristics=[]), BluetoothGATTService(uuid='0000003e-0000-1000-8000-0026bb765291', handle=11, characteristics=[BluetoothGATTCharacteristic(uuid='e604e95d-a759-4817-87d3-aa005083a0d1', handle=13, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='00000014-0000-1000-8000-0026bb765291', handle=15, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=16)]), BluetoothGATTCharacteristic(uuid='00000020-0000-1000-8000-0026bb765291', handle=18, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=19)]), BluetoothGATTCharacteristic(uuid='00000021-0000-1000-8000-0026bb765291', handle=21, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=22)]), BluetoothGATTCharacteristic(uuid='00000023-0000-1000-8000-0026bb765291', handle=24, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=25)]), BluetoothGATTCharacteristic(uuid='00000030-0000-1000-8000-0026bb765291', handle=27, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=28)]), BluetoothGATTCharacteristic(uuid='00000052-0000-1000-8000-0026bb765291', handle=30, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=31)]), BluetoothGATTCharacteristic(uuid='00000053-0000-1000-8000-0026bb765291', handle=33, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=34)]), BluetoothGATTCharacteristic(uuid='34ab8811-ac7f-4340-bac3-fd6a85f9943b', handle=36, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=37)]), BluetoothGATTCharacteristic(uuid='00000220-0000-1000-8000-0026bb765291', handle=39, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=40)])]), BluetoothGATTService(uuid='000000a2-0000-1000-8000-0026bb765291', handle=41, characteristics=[BluetoothGATTCharacteristic(uuid='e604e95d-a759-4817-87d3-aa005083a0d1', handle=43, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='000000a5-0000-1000-8000-0026bb765291', handle=45, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=46)]), BluetoothGATTCharacteristic(uuid='00000037-0000-1000-8000-0026bb765291', handle=48, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=49)])]), BluetoothGATTService(uuid='00000055-0000-1000-8000-0026bb765291', handle=50, characteristics=[BluetoothGATTCharacteristic(uuid='e604e95d-a759-4817-87d3-aa005083a0d1', handle=52, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='0000004c-0000-1000-8000-0026bb765291', handle=54, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=55)]), BluetoothGATTCharacteristic(uuid='0000004e-0000-1000-8000-0026bb765291', handle=57, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=58)]), BluetoothGATTCharacteristic(uuid='0000004f-0000-1000-8000-0026bb765291', handle=60, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=61)]), BluetoothGATTCharacteristic(uuid='00000050-0000-1000-8000-0026bb765291', handle=63, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=64)])]), BluetoothGATTService(uuid='00001530-1212-efde-1523-785feabcd123', handle=65, characteristics=[BluetoothGATTCharacteristic(uuid='e604e95d-a759-4817-87d3-aa005083a0d1', handle=67, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='000000a5-0000-1000-8000-0026bb765291', handle=69, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=70)]), BluetoothGATTCharacteristic(uuid='00001531-1212-efde-1523-785feabcd123', handle=72, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=73)]), BluetoothGATTCharacteristic(uuid='00000023-0000-1000-8000-0026bb765291', handle=75, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=76)])]), BluetoothGATTService(uuid='00000080-0000-1000-8000-0026bb765291', handle=77, characteristics=[BluetoothGATTCharacteristic(uuid='e604e95d-a759-4817-87d3-aa005083a0d1', handle=79, properties=2, descriptors=[]), BluetoothGATTCharacteristic(uuid='000000a5-0000-1000-8000-0026bb765291', handle=81, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=82)]), BluetoothGATTCharacteristic(uuid='00000023-0000-1000-8000-0026bb765291', handle=84, properties=10, descriptors=[BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=85)]), BluetoothGATTCharacteristic(uuid='0000006a-0000-1000-8000-0026bb765291', handle=87, properties=42, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=88), BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=89)]), BluetoothGATTCharacteristic(uuid='00000079-0000-1000-8000-0026bb765291', handle=91, properties=42, descriptors=[BluetoothGATTDescriptor(uuid='00002902-0000-1000-8000-00805f9b34fb', handle=92), BluetoothGATTDescriptor(uuid='dc46f0fe-81d2-4616-b5d9-6abdd796939a', handle=93)])])])
```

**Comment by bdraco** [comment]
It looks like handle allocation is unstable on the esp32?

44 is missing now

**Comment by bdraco** [comment]
handle allocation is unstable on the esp32 platforms 😢

**Comment by bdraco** [comment]
so we need to

- [ ] rework the cache in HA to be per config entry instead of per domain
- [ ] rework aiohomekit to use a combined lookup

File: aioesphomeapi/client.py [hunk 2]
```diff
@@ -485,6 +485,12 @@ def unsub() -> None:
         except asyncio.TimeoutError as err:
             unsub()
             raise TimeoutAPIError("Timeout waiting for connect response") from err
+        finally:
+            _LOGGER.warning("finished connect finally")
+            if not event.is_set():
+                _LOGGER.warning("connect failed, disconnecting")
```
Comment (bdraco) [inline aioesphomeapi/client.py:491]:
```suggestion
```

File: aioesphomeapi/client.py [hunk 3]
```diff
@@ -485,6 +485,12 @@ def unsub() -> None:
         except asyncio.TimeoutError as err:
             unsub()
             raise TimeoutAPIError("Timeout waiting for connect response") from err
+        finally:
+            _LOGGER.warning("finished connect finally")
```
Comment (bdraco) [inline aioesphomeapi/client.py:489]:
```suggestion
```

File: aioesphomeapi/client.py (hunk 1 above)
Comment (bdraco) [inline aioesphomeapi/client.py:493]:
```suggestion
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**